packages/flutter/lib/src/material/color_scheme.dart

```diff
@@ -812,12 +812,17 @@ class ColorScheme with Diagnosticable {
       'This feature was deprecated after v3.18.0-0.1.pre.',
     )
     Color? surfaceVariant,
-  }) : _primaryContainer = primaryContainer,
+  }) : primary = _lightSystemPalette?.canvasText.value ?? primary ?? const Color(0xff0000ba),
```

Comment:
`canvas` sounds more like the general background color. Should we just leave primary for now? I feel `canvas` and `canvasText` should be mapped to `surface` and `onSurface` which I saw the PR already did:)